<launch>

    <arg name="car1_name" default="car1" />
    <arg name="car2_name" default="car2" />
    <arg name="car3_name" default="car3" />
    <arg name="car4_name" default="car4" />


    <!-- Set to 1 if you want to run the map_server -->
    <arg name="map_server" value = "1"/>

    <!-- Launch  map server-->
    <group if="$(arg map_server)">
        <include file="$(find mushr_sim)/launch/map_server.launch" />
    </group>

    <!-- Launch first car -->
    <group ns="$(arg car1_name)">
         <include file="$(find mushr_sim)/launch/single_car.launch" >
            <arg name="car_name" value="$(arg car1_name)"/>

            <!-- Could be racecar-mit, racecar-uw-tx2, or racecar-uw-nano -->
            <arg name="racecar_version" value="racecar-uw-nano"/>

            <!-- The colors of the racecar, should be of the form "-<platform_color>-<inset_color>" -->
            <!-- An empty string will result in the default URDF -->
            <!-- Check CMakeLists.txt of mushr_description for appropriate values -->
            <arg name="racecar_color" value="" />
        </include>
        <node pkg="nhttc_ros" type="nhttc_node" name="controller" launch-prefix="xterm -e">
            <param name="car_name" value="car1"/>
        </node>
    </group>

    <!-- Launch second car -->
    <group ns="$(arg car2_name)">
        <include file="$(find mushr_sim)/launch/single_car.launch" >
            <arg name="car_name" value="$(arg car2_name)"/>
            <arg name="racecar_version" value="racecar-uw-nano"/>
            <arg name="racecar_color" value="-green-black" />
        </include>
        <node pkg="nhttc_ros" type="nhttc_node" name="controller">
            <param name="car_name" value="car2"/>
        </node>
    </group>

    <group ns="$(arg car3_name)">
        <include file="$(find mushr_sim)/launch/single_car.launch" >
            <arg name="car_name" value="$(arg car3_name)"/>
            <arg name="racecar_version" value="racecar-uw-nano"/>
            <arg name="racecar_color" value="-green-black" />
        </include>
        <node pkg="nhttc_ros" type="nhttc_node" name="controller3">
            <param name="car_name" value="car3"/>
        </node>
    </group>

    <group ns="$(arg car4_name)">
        <include file="$(find mushr_sim)/launch/single_car.launch" >
            <arg name="car_name" value="$(arg car4_name)"/>
            <arg name="racecar_version" value="racecar-uw-nano"/>
            <arg name="racecar_color" value="-green-black" />
        </include>
        <node pkg="nhttc_ros" type="nhttc_node" name="controller4">
            <param name="car_name" value="car4"/>
        </node>
    </group>

    <!-- Launch third car, fourth car, etc by following above template -->

</launch>
